Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.Ballistic.cs

```diff
@@ -71,18 +77,63 @@ private void OnBallisticAfterInteract(EntityUid uid, BallisticAmmoProviderCompon
             Deleted(args.Target) ||
             !TryComp<BallisticAmmoProviderComponent>(args.Target, out var targetComponent) ||
             targetComponent.Whitelist == null)
-        {
             return;
-        }
 
         args.Handled = true;
 
-        // Continuous loading
-        _doAfter.TryStartDoAfter(new DoAfterArgs(EntityManager, args.User, component.FillDelay, new AmmoFillDoAfterEvent(), used: uid, target: args.Target, eventTarget: uid)
+        ReloadDoAfter(component, args.User, args.Target.Value, args.Used);
+    }
+
+    private void OnBallisticAfterInteractUsing(EntityUid uid, BallisticAmmoProviderComponent component, AfterInteractUsingEvent args)
+    {
+        if (args.Handled ||
+            !Timing.IsFirstTimePredicted ||
```

Comment:
It seems `!Timing.IsFirstTimePredicted` is the culprit for your prediction issues? I removed it and it seemed to make the behavior work as expected.